mesop/runtime/context.py

```diff
@@ -62,9 +83,24 @@ def node_slot_children_count(self) -> int | None:
   def set_current_node(self, node: pb.Component) -> None:
     self._current_node = node
 
+  def set_previous_node_from_current_node(self) -> None:
```

Comment:
I don't think this method is being called anywhere? I tried running your PR locally, but I don't actually see any component diffs happening, I think because this method isn't actually being used.

Reply:
Doh, good catch. Looks like a regression when I added the reset_nodes method. It was supposed to call set_previous_node_from_current_node there.

Reply:
I patched your CL but I still don't see any component diffs. If you check the Chrome DevTools console logs, the protos received from the server are logged there. I don't see any component diffs in the proto.

Reply:
Ok, I think I got it working now 🤞🏾. Turns I didn't catch it because I was testing on the chat app which does the streaming response. But with the non-streaming response events it was not returning a diff. So basically the diff was not working on the first loop since previous node was not being set.

So that's the one thing I haven't figured out yet. I had to add back the original trace_mode loop code rather than existing early. However, I'm not sure why that works. I'm probably missing something pretty subtle.

Reply:
SGTM, we can debug this some more later.